proto/OptimizerConfig.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto2";
+ 
+option optimize_for = LITE_RUNTIME;
```

Comment:
this is used in SerilazieAsString function calls.
there is two version of protobuf message seriliaze header file of `message.h` `message_lite.h`
 option will declare which one to include.  Only the `message_lite.h` contains `AsString` interface, which is more clear.

https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message_lite